### PR TITLE
strawberryperl: few improvements

### PR DESCRIPTION
**Pull request by SpaceIm**
minor improvements

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/strawberryperl/all/test_package/conanfile.py [hunk 1]
```diff
@@ -1,8 +1,6 @@
-import os
 from conan import ConanFile
 from conan.tools.build import can_run
```
Comment (prince-chrismc) [inline recipes/strawberryperl/all/test_package/conanfile.py:2]:
```suggestion
```

**Comment by prince-chrismc** [comment]
> minor improvements

it's just styling 🙈 

***

Just one linter comment

File: recipes/strawberryperl/all/test_package/conanfile.py [hunk 2]
```diff
@@ -14,7 +12,6 @@ def build_requirements(self):
         self.tool_requires(self.tested_reference_str)
 
     def test(self):
-        if can_run(self):
-            self.run("perl --version")
-            perl_script = os.path.join(self.source_folder, "list_files.pl")
-            self.run(f"perl {perl_script}", env="conanbuild")
+        self.run("perl --version")
```
Comment (prince-chrismc) [inline recipes/strawberryperl/all/test_package/conanfile.py:15]: Can you please add a test for conf info? trying to get an example for the requests in #13490 

the docs say this should work in both majors 🤞 

```suggestion
        self.run("perl --version")
        conf_perl = self.conf_info.get("user.strawberryperl:perl")
        self.run(f"{conf_perl} --version")
```

**Review by jwillikers — DISMISSED** [review]

**Comment by SpaceIm** [comment]
> > minor improvements
> 
> it's just styling 🙈
> 
> Just one linter comment

nop, it avoids `from conans import __version__ as conan_version` which is not part of conan v2 API I think

File: recipes/strawberryperl/all/test_package/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/strawberryperl/all/test_package/conanfile.py:14]: it's broken in test package for conan < 1.53.0

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**